### PR TITLE
Updated matching to remove redundant python versions.

### PR DESCRIPTION
**Pull request by workhorsy**
This should remove redundant version numbers such as:
```{2, 2.6, 2.7} -> {2.6, 2.7}```
or
```{2, 3.2} -> {2, 3.2}```
or
```{3, 3.0} -> {3.0}```

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1639,6 +1639,15 @@ cache(function(data, match, sendBadge, request) {
             versions.push(matched[1]);
           }
         }
+        ['2', '3'].forEach(function(version) {
+            if (versions.some(function(element, index, array) { return new RegExp('^' + version + '\\.\\d$').exec(element); })) {
+                versions.forEach(function(element, index, array) {
+                    if (element === version) {
+                        array.splice(index, 1);
```
Comment (espadrine) [inline server.js]: I'm not a fan of looping over something that changes under our feet. Could we filter `versions` and assign the new list to `versions` afterwards?

**Comment by workhorsy** [comment]
That should do it.

**Comment by espadrine** [comment]
Thanks for your contribution!

I tweaked it a bit here for simplicity, without modifying the algorithm: https://github.com/badges/shields/commit/cb2b206723916e3257a76a9efd5abdfe4f3ccb5d

**Comment by xflr6** [comment]
Thanks!

**Pull request merged**